src/rdkafka_cgrp.c

```diff
@@ -3238,17 +4489,49 @@ rd_kafka_cgrp_subscribe (rd_kafka_cgrp_t *rkcg,
         if (rkcg->rkcg_rk->rk_conf.enabled_assignor_cnt == 0)
                 return RD_KAFKA_RESP_ERR__INVALID_ARG;
 
+        /* If the consumer has raised a fatal error treat all subscribes as
+           unsubscribe */
+        if (rd_kafka_fatal_error_code(rkcg->rkcg_rk)) {
+                if (rkcg->rkcg_subscription)
+                        rd_kafka_cgrp_unsubscribe(rkcg, 1/*leave group.*/);
+                return RD_KAFKA_RESP_ERR__FATAL;
+        }
+
+        if (RD_KAFKA_CGRP_REBALANCING(rkcg) &&
+            (rd_kafka_cgrp_rebalance_protocol(rkcg) ==
+             RD_KAFKA_REBALANCE_PROTOCOL_COOPERATIVE)) {
+                /* TODO: Postponing subscribe is not implemented for the
+                 *       EAGER case, but this may be relevant. */
+                rd_kafka_dbg(rkcg->rkcg_rk, CGRP|RD_KAFKA_DBG_CONSUMER,
+                             "SUBSCRIBE", "Group \"%.*s\": postponing "
+                             "subscribe until previous rebalance "
+                             "completes (join state %s)",
+		             RD_KAFKAP_STR_PR(rkcg->rkcg_group_id),
+                             rd_kafka_cgrp_join_state_names[rkcg->rkcg_join_state]);
+                if (rkcg->rkcg_next_subscription)
+                        rd_kafka_topic_partition_list_destroy_free(
+                                rkcg->rkcg_next_subscription);
+                rkcg->rkcg_next_subscription = NULL;
+
+                if (!rktparlist)
+                        rkcg->rkcg_next_unsubscribe = rd_true;
+                else
+                        rkcg->rkcg_next_subscription = rktparlist;
```

Comment:
This should set rkcg_next_unsubscribe to false

Reply:
good point. I put it together with `rkcg_next_subscription` above the if to make it logically a "clear next" operation.